### PR TITLE
Translations

### PR DESCRIPTION
**Pull request by mimiflynn**
Addresses issue #140 

This PR is so large because of the number of files moved into a new directory to support translations. We will need to further review and adjust the content to account for translations support living within the `/exercises/`directory. Because the translations will not be a 1:1 for each curriculum, each language index will be honed to the existing content. I'd like to automate this page, but that will have to live in another PR.

### CODE REVIEW

File: site/content/exercises/en-US/circuitpython/index.mdx
Comment (mimiflynn) [inline site/content/exercises/en-US/circuitpython/index.mdx:1]: This file replaces what used to be a dynamic page in the `/pages/` directory.  Ideally, all content will live in  `/content/` and all content management will live in `/src/`.

File: site/gatsby-node.js [hunk 1]
```diff
@@ -41,11 +41,18 @@ exports.createPages = async ({ graphql, actions }) => {
...
-  const exerciseTemplate = path.resolve(`./src/templates/exercise.js`);
-  const pageTemplate = path.resolve(`./src/templates/page.js`);
+  const exerciseTemplate = path.resolve(`./src/templates/exercise.jsx`);
+  const indexTemplate = path.resolve(`./src/templates/index.jsx`);
+  const pageTemplate = path.resolve(`./src/templates/page.jsx`);
 
   function getTemplate(page) {
-    return page.frontmatter.exercise ? exerciseTemplate : pageTemplate;
+    const path = page.internal.contentFilePath;
+    const isExerciseIndex = path.includes('index') && path.includes('exercise');
```
Comment (mimiflynn) [inline site/gatsby-node.js:50]: Creates a new page template for `index.mdx` files that live in `/exercises/`.

File: site/package.json [hunk 2]
```diff
@@ -46,10 +46,10 @@
     "serve": "gatsby serve",
     "clean": "gatsby clean",
     "test": "echo \"Write tests! -> https://gatsby.dev/unit-testing\" && exit 1",
-    "pub": "rm -rf ../docs && mv public ../docs",
-    "pubWindows": "rd /S \"../docs\" && move public \"../docs\""
+    "pub": "rimraf ../docs && mv public ../docs"
```
Comment (mimiflynn) [inline site/package.json:49]: `rimraf` for all the OS's.

File: site/src/constants/mdx-shortcodes.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Link } from 'gatsby';
+import Hero from '../components/hero';
+import SummaryCard from '../components/summary-card';
+import Translations from '../components/translations';
+
+export const SHORTCODES = { Hero, Link, SummaryCard, Translations };
```
Comment (mimiflynn) [inline site/src/constants/mdx-shortcodes.js:6]: These are the React components that are available in `.mdx` files.

[Shortcodes](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#shortcodes)

File: site/src/pages/404.js [hunk 4]
```diff
@@ -2,25 +2,48 @@ import React from 'react';
...
     <Layout location={location}>
-      <h1>Not Found</h1>
-      <p>You just hit a route that doesn&#39;t exist... the sadness.</p>
+      <div className="content">
+        <h2>404 Not Found</h2>
+        <p>
+          Sorry, the url you're looking for appears to be missing. Check out the
+          sitemap below to find your expected page.
+        </p>
+        <SiteMap location={location} nodes={nodes} />
```
Comment (mimiflynn) [inline site/src/pages/404.js:19]: With the url updates, I went ahead and enhanced the 404 page with a sitemap so a user could hopefully find what they were looking for.

File: site/src/components/head.jsx
Comment (mimiflynn) [inline site/src/components/head.jsx:1]: `Head` component to be used with [Gatsby Head API](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/).

File: site/src/pages/404.js (hunk 4 above)
Comment (kristen-holmes) [reply site/src/pages/404.js:19]: I liked the sadness

Comment (mimiflynn) [reply site/src/pages/404.js:19]: Sadness is always available.

**Review by kristen-holmes — APPROVED** [review]

**Review by olivierbeau — APPROVED** [review]
Seems good from a high level perspective.

@mimiflynn Do we have a QA version of the website where we can preview changes in a commit ?

If not, it might be a cool addition :)

**Comment by mimiflynn** [comment]
@olivierbeau I was just telling @kristen-holmes that I wish we had PR previews to act as QA but we don't have the option at the moment.

**Pull request merged**